js/src/lib/idna.js

```diff
@@ -8,7 +8,7 @@ const PUNYCODE_ENCODED_DOMAIN_PREFIX = 'xn--';
 // See https://tools.ietf.org/html/rfc3490#section-4.1 for details
 const idna = {
   toAscii: function(domain) {
-    if ((domain.substring(0, 4) === PUNYCODE_ENCODED_DOMAIN_PREFIX) && !domain.match(validAsciiDomain)) {
+    if ((domain.indexOf(PUNYCODE_ENCODED_DOMAIN_PREFIX) === 0) && !domain.match(validAsciiDomain)) {
```

Comment:
I think we want to check for the prefix only at the beginning of the string. I think substring works fine for this use case. Any other concerns?

Reply:
@kaushlakers substring works fine here too. I see you pushed a stand-alone version for 2.0.3. I'll give that a try and let you know.

Reply:
@kaushlakers 2.0.3 of the library does not exist in the tag 2.0.3